nrf_modem/include/nrf_modem_os_rpc.h

```diff
@@ -0,0 +1,246 @@
+		uintptr_t addr;
+		/** TX shared memory size. */
+		size_t size;
+	} tx;
+	struct {
+		/** Local signal device. */
+		uintptr_t sigdev;
+		/** RX channel number. */
+		uint32_t ch;
+		/** RX shared memory start address. */
+		uintptr_t addr;
+		/** RX shared memory size. */
+		size_t size;
+	} rx;
+	struct {
+		/** Pointer to private context passed as an argument to callbacks. */
+		void *priv;
+		/**
+		 * @brief Bind was successful.
+		 *
+		 * Invoked when the local and remote rpc instances
+		 * have bound successfully.
+		 *
+		 * @param priv Private user data.
+		 */
+		void (*bound)(void *priv);
+		/**
+		 * @brief New message has arrived.
+		 *
+		 * Invoked when new data is received.
+		 *
+		 * @param data Pointer to data buffer.
+		 * @param len Length of @a data.
+		 * @param priv Private user data.
+		 */
+		void (*received)(const void *data, size_t len, void *priv);
+	} cb;
+};
+
+/**
+ * @brief Parameters for configuring an nrf_modem signaling instance.
+ *
+ * Passed when invoking @c nrf_modem_os_rpc_signal_init.
+ */
+struct nrf_modem_os_rpc_signal_config {
+	/** Signal device. */
+	uintptr_t sigdev;
+	/** Channel number. */
+	uint32_t ch;
+	/** Pointer to private context passed as an argument to callbacks. */
+	void *priv;
+	/** Invoked when signal with channel num @a ch is raised.
+	 *  See @c nrf_modem_os_rpc_signal_cb_t.
+	 */
+	nrf_modem_os_rpc_signal_cb_t recv;
+};
+
+/**
+ * @brief RPC instance declaration used internally in nrf_modem.
+ *
+ * This structure must be defined in the glue.
+ */
+struct nrf_modem_os_rpc;
+
+/**
+ * @brief Signaling instance declaration used internally in nrf_modem.
+ *
+ * This structure must be defined in the glue.
+ */
+struct nrf_modem_os_rpc_signal;
+
+/**
+ * @brief RPC instances used by nrf_modem and defined externally.
+ */
+extern struct nrf_modem_os_rpc inst_ctrl;
+extern struct nrf_modem_os_rpc inst_data;
+
+/**
+ * @brief Signaling instances used by nrf_modem and defined externally.
+ */
+extern struct nrf_modem_os_rpc_signal inst_app_fault;
+extern struct nrf_modem_os_rpc_signal inst_modem_fault;
+extern struct nrf_modem_os_rpc_signal inst_modem_sysoff;
+
+/**
+ * @brief Get address of the application signal device.
+ *
+ * @return Address of application signal device.
+ */
+uintptr_t nrf_modem_os_rpc_sigdev_app_get(void);
+
+/**
+ * @brief Get address of the modem signal device.
+ *
+ * @return Address of modem signal device.
+ */
+uintptr_t nrf_modem_os_rpc_sigdev_modem_get(void);
+
+/**
+ * @brief Open an RPC instance.
+ *
+ * @param instance Pointer to RPC instance.
+ * @param conf Parameters for configuring the instance before it is opened.
+ *
+ * @return 0 on success, a negative errno otherwise.
+ */
+int nrf_modem_os_rpc_open(struct nrf_modem_os_rpc *instance,
+			  const struct nrf_modem_os_rpc_config *conf);
+
+/**
+ * @brief Send a message from the local RPC instance to the remote instance.
+ *
+ * @param instance Pointer to RPC instance.
```

Comment:
Which RPC instance is this one? The local or the remote one?

Reply:
Thanks. I see that the word 'instance' have been used with multiple meaning. I have made some improvements.